src/Kros.AspNetCore/Authorization/ServiceCollectionExtensions.cs

```diff
@@ -21,12 +24,42 @@ public static class ServiceCollectionExtensions
     /// Configure gateway authorization.
     /// </summary>
     /// <param name="services">Collection of app services.</param>
-    public static IServiceCollection AddGatewayJwtAuthorization(this IServiceCollection services)
-        => services
-        .AddMemoryCache()
-        .AddHttpClient(GatewayAuthorizationMiddleware.AuthorizationHttpClientName)
-        .Services;
+    /// <param name="configuration">Configuration.</param>
+    public static IServiceCollection AddGatewayJwtAuthorization(
+        this IServiceCollection services,
+        IConfiguration configuration)
+    {
+        services.ConfigureGatewayJwtAuthorization(configuration)
+            .WithMemoryCache();
+            
+        return services
+            .AddMemoryCache();
+    }
 
+    /// <summary>
+    /// Configure gateway authorization.
+    /// </summary>
+    /// <param name="services">Collection of app services.</param>
+    /// <param name="configuration">Configuration.</param>
+    /// <returns>Gateway JWT authorization registrator.</returns>
+    public static GatewayJwtAuthorizationRegistrator ConfigureGatewayJwtAuthorization(
```

Comment:
Túto metódu by som dal `private` a metódu `AddGatewayJwtAuthorization` by som spravil tak aby vracala `GatewayJwtAuthorizationRegistrator`. Následne v metóde `WithHybridCache` by som nepoužíval `.AddScope` ale `.Replace`. Tým pádom môžeš pekne spraviť to čo robia ostatné frameworky, že defult použie je `AddGatewayJwtAuthorization()` a keď chce niekto niečo iné tak spraví `.services.AddGatewayJwtAuthorization().WithHybridCache()`.